File: Занятие 1 - Сложность, тестирование, особые случаи/A. Покраска деревьев.py
p, v = map(int, input().split())
q, m = map(int, input().split())

v1 = p-v
v2 = p+v
m1 = q-m
m2 = q+m 

if min(v2, m2) < max(v1, m1):
    print(v2 - v1 + 1 + m2 - m1 + 1)
else:
    print(max(v2, m2) - min(v1, m1) + 1)

File: Занятие 1 - Сложность, тестирование, особые случаи/A_Покраска_деревьев.cpp
#include <iostream>

using namespace std;

int main() {
    int p, v;
    int q, m;

    cin >> p >> v;
    cin >> q >> m;

    int v1 = p - v;
    int v2 = p + v;
    int m1 = q - m;
    int m2 = q + m;

    if(min(v2, m2) < max(v1, m1)) {
        cout << (v2 - v1 + 1 + m2 - m1 + 1);
    }
    else
        cout << (max(v2, m2) - min(v1, m1) + 1);

    return 0;
}
